File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
         if(x == 0){
            return true;
        }if (x < 0|| x % 10 == 0){
            return false;
        }
        int y = 0;
        while(x > y){
            int pop = x % 10;
            x/=10;
            y =(y * 10)+ pop;
        }
        if (x == y || x == y/10){
            return true;
        }else{
            return false;
        }
    }
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        r=0
        temp=x
        while x>0:#1
            r=(x%10)+r*10  #r=1+54320  
            x=x//10 #x=0 r=54321
        return (r==temp)




